### PR TITLE
[MXNET-16] move dmlc-core & nnvm

### PR DESCRIPTION
**Pull request by eric-haibin-lin**
#10213 

## Description ##
(Brief description on what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by eric-haibin-lin** [comment]
@mbaijal could you review license changes? I noticed that two files under src/nnvm didn't have the apache header

**Comment by szha** [comment]
@mbaijal please review the licenses and patch if necessary.

File: LICENSE [hunk 1]
```diff
@@ -220,13 +220,13 @@
...
     5. 3rdparty/dlpack - For details, see, 3rdparty/dlpack/LICENSE
-    6. dmlc-core - For details, see, dmlc-core/LICENSE
+    6. 3rdparty/dmlc-core - For details, see, 3rdparty/dmlc-core/LICENSE
     7. 3rdparty/mshadow - For details, see, 3rdparty/mshadow/LICENSE
-    8. nnvm/dmlc-core - For details, see, nnvm/dmlc-core/LICENSE
-    9. nnvm - For details, see, nnvm/LICENSE
-    10. nnvm-fusion - For details, see, nnvm/plugin/nnvm-fusion/LICENSE
+    8. 3rdparty/nnvm/dmlc-core - For details, see, 3rdparty/nnvm/dmlc-core/LICENSE
+    9. 3rdparty/nnvm - For details, see, 3rdparty/nnvm/LICENSE
+    10. nnvm-fusion - For details, see, 3rdparty/nnvm/plugin/nnvm-fusion/LICENSE
```
Comment (mbaijal) [inline LICENSE]: following the same pattern, you can add the 3rdparty path for nnvm-fusion, but I guess this should also do

Comment (eric-haibin-lin) [reply LICENSE]: Updated in https://github.com/apache/incubator-mxnet/pull/10370

**Pull request merged**